src/Blazor.FileReader/script/DragnDrop.ts

```diff
@@ -44,23 +45,126 @@ function BuildDragEventHandler(declaredMethod: string, script: string, eventDesc
     return (() => { }) as DragEventHandler;
 }
 
+// Go through the data transfer object to pull out the files and folders
+async function getFilesAsync(dataTransfer: DataTransfer): Promise<FileList> {
+    //return await new Promise<FileList>((resolve, reject) => {
+    const files: File[] = [];
+    const len = dataTransfer.items.length;
+    const webkitQueue: FileSystemEntry[] = [];
+    const fileQueue: File[] = [];
+
+    // First must save off all the files
+    for (let i = 0; i < len; i++) {
+        const item = dataTransfer.items[i];
+        if (item.kind === "file") {
+            if (typeof item.webkitGetAsEntry === "function") {
+                const entry = item.webkitGetAsEntry();
+                webkitQueue.push(entry);
+            } else {
+                const file = item.getAsFile();
+                if (file) {
+                    fileQueue.push(file);
+                }
+            }
+        }
+    }
+
+    // Get the content for the web kit files
+    for (let i = 0; i < webkitQueue.length; i++) {
+        const file = await readEntryAsync(webkitQueue[i]);
+        files.push(...file);
+    }
+
+    // Get the content for any other files
+    files.push(...fileQueue);
+
+    return new FileEntryList(files);
+}
+
+async function readEntryAsync(innerEntry: FileSystemEntry): Promise<File[]> {
+    const files: File[] = [];
+
+    if (isFile(innerEntry)) {
+        let fullPath = innerEntry.fullPath;
+        if (fullPath.charAt(0) === "/" || fullPath.charAt(0) === "\\")
+            fullPath = fullPath.substring(1);
+        try {
+            const file = await getFile(innerEntry);
+            files.push(redefineWebkitRelativePath(file, fullPath));
+        } catch (err) {
+            console.error(`error on ${fullPath}`);
+            console.error(err);
+        }
+    } else if (isDirectory(innerEntry)) {
+        try {
+            const entries = await getEntries(innerEntry.createReader());
+            for (const entry of entries) {
+                const innerFiles = await readEntryAsync(entry);
+                files.push(...innerFiles);
+            }
+        } catch (err2) {
+            console.error(err2);
+        }
+    }
+
+    return files;
+}
+
+async function getEntries(reader): Promise<FileSystemFileEntry[]> {
+    try {
```

Comment:
Same here, remove the try-catch

Reply:
Everything else looks great and tests seems to work well. I'll merge and release a new version once the try-catch's are removed. Thanks for the contribution and your patience!

Reply:
Done Removed